spec/lib/wifi_user/use_cases/email_signup_spec.rb

```diff
@@ -112,6 +116,39 @@
           expect(notify_email_stub).to have_been_requested.times(1)
         end
       end
+
+      context "given an email address with a non-gov domain, but email is sponsored" do
+        let(:allowlist_checker) { double(execute: { success: false }) }
+        let(:sponsee_is_user_checker) { double(execute: true) }
+        let!(:notify_template_id) { "4f6bae31-5add-43a9-b0e3-7db43452676e" }
+        let!(:sponsored_user) { FactoryBot.create(:user_details, :recent, :active, :sponsored) }
+        let!(:created_contact) { sponsored_user.contact }
+        let!(:sponsor) { sponsored_user.sponsor }
+        let!(:username) { sponsored_user.contact }
+        let!(:password) { sponsored_user.password }
+        let(:notify_email_request) do
+          {
+            email_address: created_contact,
+            template_id: notify_template_id,
+            personalisation: {
+              username:,
+              password:,
+              sponsor:,
+            },
+            email_reply_to_id: do_not_reply_id,
+          }
+        end
+
+        it "returns credentials username and password" do
```

Comment:
Not sure if we are bothered whether `user_model` receives `generate` or not? The only thing this is supposed to do is send an email right?

Reply:
On line 109 we are testing`expect(user_model).not_to receive(:generate)` given User is non-gov.
I assumed we want to test the opposite if User is Sponsored?

Reply:
This test is supposed to test that an email is sent when a sponsored user sends a message right? The `User.generate` method is basically a `find_or_create` call and I think it may be a bit of an implementation detail if there is code present that finds a user. If you want to test that no additional user is being created it may be easier to do

```
expect {
  subject.execute(contact: created_contact)
}.to_not change { User.count }
```